src/istgt_lu.c

```diff
@@ -1737,11 +1757,46 @@ istgt_lu_add_unit(ISTGT_Ptr istgt, CF_SECTION *sp)
 	ISTGT_TRACELOG(ISTGT_TRACE_DEBUG, "ConsistencyFactor %d\n",
 	    lu->consistency_factor);
 
+	if (lu->replication_factor > lu->desired_replication_factor) {
+		ISTGT_ERRLOG("Invalid config ReplicationFactor(%d) is"
+		    " greater than Desired ReplicationFactor(%d)\n",
+		    lu->replication_factor, lu->desired_replication_factor);
+		goto error_return;
+	}
 	if (lu->replication_factor <= 0 || lu->consistency_factor <= 0 ||
 		lu->replication_factor < lu->consistency_factor) {
 		ISTGT_ERRLOG("Invalid ReplicationFactor/ConsistencyFactor or their ratio\n");
 		goto error_return;
 	}
+	/* Read trusty replica details from conf file and maintain in memory structure*/
+	val = istgt_get_val(sp, "Replica");
+	if (val == NULL) {
+		TAILQ_INIT(&lu->trusty_replicas);
+	} else {
+		TAILQ_INIT(&lu->trusty_replicas);
+		for (i=0; ; i++) {
+			key = istgt_get_nmval(sp, "Replica", i, 0);
```

Comment:
@payes can you please review istgt related code for config parsing, sock files etc